frontend/components/q&a/useQaTabs.ts

```
@@ -124,19 +103,26 @@ export function useQaTabs() {
         );
         setTotalPages(data.totalPages);
       } catch (error) {
+        if (!isActive || controller.signal.aborted) {
+          return;
+        }
         console.error('Failed to load questions:', error);
         setItems([]);
         setTotalPages(0);
       } finally {
+        if (!isActive) {
+          return;
```

Biome (2.1.2)

[error] Unsafe usage of 'return'.

'return' in 'finally' overwrites the control flow statements inside 'try' and 'catch'.

(lint/correctness/noUnsafeFinally)